src/main/java/de/tum/cit/aet/artemis/communication/web/FaqResource.java

```diff
@@ -0,0 +1,179 @@
+     */
+    @GetMapping("faqs/{faqId}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Faq> getFaq(@PathVariable Long faqId) {
+        log.debug("REST request to get faq {}", faqId);
+        Faq faq = faqRepository.findByIdElseThrow(faqId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, faq.getCourse(), null);
+        return ResponseEntity.ok(faq);
+    }
+
+    /**
+     * DELETE /faqs/:faqId : delete the "id" faq.
+     *
+     * @param faqId the id of the faq to delete
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("faqs/{faqId}")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Void> deleteFaq(@PathVariable Long faqId) {
+
+        log.debug("REST request to delete faq {}", faqId);
+        Faq faq = faqRepository.findByIdElseThrow(faqId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, faq.getCourse(), null);
+        faqRepository.deleteById(faqId);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, faqId.toString())).build();
+    }
+
+    /**
+     * GET /courses/:courseId/faqs : get all the faqs of a course
+     *
+     * @param courseId the courseId of the course for which all faqs should be returned
+     * @return the ResponseEntity with status 200 (OK) and the list of faqs in body
+     */
+    @GetMapping("courses/{courseId}/faqs")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Set<Faq>> getFaqForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get all Faqs for the course with id : {}", courseId);
+
+        Course course = getCourseForRequest(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
+        Set<Faq> faqs = faqRepository.findAllByCourseId(courseId);
+        return ResponseEntity.ok().body(faqs);
+    }
+
+    /**
+     * GET /courses/:courseId/faq-categories : get all the faq categories of a course
+     *
+     * @param courseId the courseId of the course for which all faq categories should be returned
+     * @return the ResponseEntity with status 200 (OK) and the list of faqs in body
+     */
+    @GetMapping("courses/{courseId}/faq-categories")
+    @EnforceAtLeastStudent
+    public ResponseEntity<Set<String>> getFaqCategoriesForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get all Faq Categories for the course with id : {}", courseId);
+
+        Course course = getCourseForRequest(courseId);
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
+        Set<String> faqs = faqRepository.findAllCategoriesByCourseId(courseId);
+
+        return ResponseEntity.ok().body(faqs);
+    }
+
+    /**
+     *
+     * @param courseId the courseId of the course
+     * @return the course with the id courseId, unless it exists
+     */
```

Comment:
Can you double-check this doc comment?

- You should describe the method
- I think you mean "unless it does not exist" for the return value.